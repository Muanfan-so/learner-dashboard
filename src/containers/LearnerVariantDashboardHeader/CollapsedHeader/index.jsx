import React from 'react';

import { useIntl } from '@edx/frontend-platform/i18n';
import { Menu, Close } from '@edx/paragon/icons';
import { IconButton, Icon } from '@edx/paragon';

import { useLearnerVariantDashboardHeaderData, useIsCollapsed } from '../hooks';

import CollapseMenuBody from './CollapseMenuBody';
import BrandLogo from '../BrandLogo';

import messages from '../messages';

export const CollapsedHeader = () => {
  const { formatMessage } = useIntl();
  const isCollapsed = useIsCollapsed();
  const { isOpen, toggleIsOpen } = useLearnerVariantDashboardHeaderData();

  return (
    isCollapsed && (
      <>
        <header className="d-flex shadow-sm align-items-center learner-variant-header">
          <IconButton
            invertColors
            isActive
            src={isOpen ? Close : Menu}
            iconAs={Icon}
            alt={
              isOpen
                ? formatMessage(messages.collapseMenuOpenAltText)
                : formatMessage(messages.collapseMenuClosedAltText)
            }
            onClick={toggleIsOpen}
            variant="primary"
            className="p-4"
          />
          <BrandLogo />
        </header>
        <CollapseMenuBody isOpen={isOpen} />
      </>
    )
  );
};

CollapsedHeader.propTypes = {};

export default CollapsedHeader;
