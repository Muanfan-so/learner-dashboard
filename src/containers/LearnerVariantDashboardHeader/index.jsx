import React from 'react';

import MasqueradeBar from 'containers/MasqueradeBar';
import ConfirmEmailBanner from 'containers/LearnerDashboardHeader/ConfirmEmailBanner';

import CollapsedHeader from './CollapsedHeader';
import ExpandedHeader from './ExpandedHeader';

import './index.scss';

export const LearnerVariantDashboardHeader = () => (
  <>
    <ConfirmEmailBanner />
    <CollapsedHeader />
    <ExpandedHeader />
    <MasqueradeBar />
  </>
);

LearnerVariantDashboardHeader.propTypes = {};

export default LearnerVariantDashboardHeader;
