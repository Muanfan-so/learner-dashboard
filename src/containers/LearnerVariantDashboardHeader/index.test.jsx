import { shallow } from 'enzyme';
import LearnerVariantDashboardHeader from '.';

jest.mock('containers/LearnerDashboardHeader/ConfirmEmailBanner', () => 'ConfirmEmailBanner');
jest.mock('containers/MasqueradeBar', () => 'MasqueradeBar');
jest.mock('./CollapsedHeader', () => 'CollapsedHeader');
jest.mock('./ExpandedHeader', () => 'ExpandedHeader');

describe('LearnerVariantDashboardHeader', () => {
  test('render', () => {
    const wrapper = shallow(<LearnerVariantDashboardHeader />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('ConfirmEmailBanner')).toHaveLength(1);
    expect(wrapper.find('MasqueradeBar')).toHaveLength(1);
    expect(wrapper.find('CollapsedHeader')).toHaveLength(1);
    expect(wrapper.find('ExpandedHeader')).toHaveLength(1);
  });
});
