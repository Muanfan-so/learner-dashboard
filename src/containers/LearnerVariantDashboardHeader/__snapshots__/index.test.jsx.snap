// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearnerVariantDashboardHeader render 1`] = `
<Fragment>
  <ConfirmEmailBanner />
  <CollapsedHeader />
  <ExpandedHeader />
  <MasqueradeBar />
</Fragment>
`;
